staging/src/k8s.io/kubectl/pkg/cmd/apply/apply.go

```diff
@@ -540,6 +543,53 @@ See https://kubernetes.io/docs/reference/using-api/server-side-apply/#conflicts`
 			return err
 		}
 
+		// Check if the returned object needs to have its
+		// kubectl-client-side-apply managed fields migrated server-side-apply.
+		//
+		// By checking afterward instead of fetching the object beforehand and
```

Comment:
Avoiding extra retrievals in the common case (no migration) is an excellent approach. And two extra REST calls for the uncommon migration case seems like a very palatable price to pay.